### ISSUE TITLE
Eliminate Data Tab

### ISSUE THREAD

**Issue by darrylthom** [luusluus/vscode-parquet-visualizer#123]
I am not sure whether or not there is still value in having the Data View. In fact, because it automatically queries my data source it ends up taking longer to getting to the querying I would actually need to do. Query View seems to serve all of the functionality Data has and more. You could even add an option to automatically execute the default query in Settings, preferably automatic=False by default)

**Comment by luusluus** [comment]
> because it automatically queries my data source it ends up taking longer to getting to the querying I would actually need to do.

I'm not sure I understand what you mean with this. Can you clarify?

Currently the data view queries the file directly, without storing it in memory. Every action, changing page size, going to a page does a new query on the file directly. It doesnt have a query editor obstructing the table.

However, the query view queries the file, and then stores the query result into memory, on which can be paginated, sorted (TBD) and searched (TBD). If i wanted to go to the last page i would need to query that, or do select * from data and click the last page button, whereas in data view id just need click last page.

I'm open to merging these two views as they're quite similar. But other users might have different use cases as well. I know that the data view currently is used much more than the query view.

**Comment by darrylthom** [comment]
For example, I am opening a 100 MB parquet file on a Network Share. When I open, the query for the Data view automatically is executing and downloading all columns and however many roles is my default page size from my file. Because this is a big file on a network file, there is a lot of time for it to download that information. And the Data view itself won’t allow me to do anything helpful to the file, it is just causing. A delay before my analysis basically.

Just because Data is the “most used”, the telematics may be misleading. A user might open the file and see that the Data tab has no useful functionality (like exporting) and not know the Query tab has the options they need. This may mean that they uninstall the plugin and try another parquet plugin. The data tab is just the default tab in the plugin and might actually be choosing users to go with alternative plugins.

**Comment by luusluus** [comment]
I see, thanks for clearing that up. You would like the query view as the default view? And remove the data view. Also provide the option/config to automatically query the data or not.

**Comment by darrylthom** [comment]
Yes, query could be default view and you could make it automatic run True by default with a config option to set it to False within the settings for edge cases like mine. If you’re worried about screen real estate, you could add collapse/expand query functionality.

**Comment by luusluus** [comment]
I'll make this my next priority. I will keep the data tab, however. 
This MR should contain:
- put query view default
- add automatic run query view boolean configuration
- add fetching data data view in separate thread in background

**Comment by darrylthom** [comment]
> I'll make this my next priority. I will keep the data tab, however. This MR should contain:
> 
> * put query view default
> * add automatic run query view boolean configuration
> * add fetching data data view in separate thread in background

This sounds like a good solution.

**Comment by luusluus** [comment]
I've made the query tab the first tab. 
I've added a setting `runQueryOnStartup` to prevent running query on startup. 
The data for data tab will be loaded asynchronously in a separate thread after vscode document has initialized.
